examples/systemcalls/systemcalls.c

```diff
@@ -92,6 +108,30 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+    int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+
+    pid = fork();
+    if(pid == -1){ 
+        return false;
+    }
+    else if(pid == 0) {
+        if(dup2(fd, 1) < 0) exit(-1);
+        close(fd);
+
+        status = execv(command[0], command);
+
+        exit(-1);
+    }
+
+    close(fd);
+
+    while(wait(&status) != pid);
+    
+    // Check if we did not exit normally OR if we exited with a non zero status
+    if(!WIFEXITED(status) || WEXITSTATUS(status)){
+        va_end(args);
+        return false;
+    }
```

Comment:
@jbohn3353 think about how you could avoid duplicating the logic between do_exec and do_exec_redirect